### PR TITLE
Add table insertion dialog feature

### PR DESCRIPTION
**Pull request by MichaelBernhardt**
## Summary

This PR adds a table insertion dialog feature to PyMarkdownEditor with the following capabilities:

- Configurable number of rows and columns (1-50 rows, 1-20 columns)
- Optional header row with auto-generated column labels
- Column alignment options (left, center, right)
- Keyboard shortcut: Ctrl+Shift+T
- Smart cursor positioning after insertion

## Implementation Details

- New `TableDialog` class with port/adapter pattern for editor integration
- Integration with main window toolbar and menu
- Generates proper Markdown table syntax with alignment markers
- Cursor automatically positioned in first editable cell

## Test Coverage

Added comprehensive test suite with 14 test cases covering:
- Dialog initialization and default values
- Table generation with various configurations
- Alignment options (left, center, right)
- Header row inclusion/exclusion
- Edge cases (minimum/maximum sizes)
- Cursor positioning after insertion
- Integration with existing editor content

## Files Changed

- `pymd/services/ui/table_dialog.py` - New table dialog implementation (188 lines)
- `pymd/services/ui/main_window.py` - Integration with main window (9 lines)
- `tests/test_table_dialog.py` - Comprehensive test suite (242 lines)

### CODE REVIEW

File: pymd/services/ui/main_window.py [hunk 1]
```diff
@@ -164,6 +166,7 @@ def _build_actions(self):
         self.act_list = QAction("List", self, triggered=lambda: self._prefix_line("- "))
         self.act_img = QAction("Image", self, triggered=lambda: self._select_image())
         self.act_link = QAction("Link", self, triggered=lambda: self._create_link())
+        self.act_table = QAction("Table", self, shortcut="Ctrl+Shift+T", triggered=self._insert_table)
```
Comment (clintonshane84) [inline pymd/services/ui/main_window.py]: `Ctrl+Shift+T` is hardcoded and on macOS this won’t work for ⌘⇧T. To keep this cross-platform behavior, either document it or map via settings. There’s no StandardKey for “Insert Table”, so consider making it configurable or omit the default shortcut.
